### PR TITLE
feat: node and yarn installer to succeed volta

### PR DESCRIPTION
**Pull request by joshuarli**
for:
https://github.com/getsentry/sentry/pull/75554
https://github.com/getsentry/getsentry/pull/15082
https://github.com/getsentry/sentry-options-automator/pull/2024

### CODE REVIEW

File: devenv/lib/archive.py [hunk 1]
```diff
@@ -20,17 +22,34 @@ def atomic_replace(src: str, dest: str) -> None:
     os.replace(src, dest)
 
 
-def download(url: str, sha256: str, dest: str = "") -> str:
+def download(
```
Comment (joshuarli) [inline devenv/lib/archive.py:25]: should i add some unittests for this? integration won't exercise retries generally

Comment (joshuarli) [reply devenv/lib/archive.py:25]: will be adding a unittest and separate pr

File: ci/devenv-bootstrap.sh [hunk 2]
```diff
@@ -21,9 +21,43 @@ export DEVENV_FETCH_BRANCH=master
 devenv bootstrap
 devenv fetch sentry
 
-cd "$HOME/code/sentry"
+# overwrite sentry's devenv config so we don't break when upstream config is updated
+cat <<EOF > "$HOME/code/sentry/devenv/config.ini"
+[venv.sentry]
+python = 3.11.8
```
Comment (asottile-sentry) [inline ci/devenv-bootstrap.sh:27]: well this is out of date now :)

**Review by asottile-sentry — APPROVED** [review]
python parts seem fine -- not sure what the shell changes are though?

**Comment by joshuarli** [comment]
> python parts seem fine -- not sure what the shell changes are though?
> 
>  [ 68747470733a2f2f692e666c756666792e63632f364b5678547368784d73727442384d6a5a374d48766c7642434a6a66746774502e676966 ](https://camo.githubusercontent.com/2f3443f5f57e94310b336cd66d7ca304fdcc6cf79cdbf49e222decf6295927dc/68747470733a2f2f692e666c756666792e63632f364b5678547368784d73727442384d6a5a374d48766c7642434a6a66746774502e676966) [ ](https://camo.githubusercontent.com/2f3443f5f57e94310b336cd66d7ca304fdcc6cf79cdbf49e222decf6295927dc/68747470733a2f2f692e666c756666792e63632f364b5678547368784d73727442384d6a5a374d48766c7642434a6a66746774502e676966)

i was trying to look into why `direnv allow` wasn't doing anything in CI and along that path i found we weren't exactly reproducing what people do after `fetch` which instructs `Please close this terminal window and start a fresh one.`, hence 
the `bash --login ci/devenv-post-bootstrap.sh`

umm, i think i can probably roll back those changes though since i forgot `install-devenv.sh` starts a new login shell after it's done

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**